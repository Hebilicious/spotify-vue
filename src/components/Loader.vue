<template>
    <div class="loader-container">
        <div class="loader"></div>
    </div>
</template>

<script lang="ts">
    export default {

    }
</script>

<style lang="scss">
.loader-container {
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    .loader {
        border: 8px solid rgba($color: #1fdf64, $alpha: 0.2); /* Light grey */
        border-top: 8px solid var(--color-accent-primary); /* Blue */
        border-radius: 50%;
        width: 60px;
        height: 60px;
        animation: spin 2s linear infinite;
    }
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}
</style>