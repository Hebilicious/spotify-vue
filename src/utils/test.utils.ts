import { VueWrapper } from '@vue/test-utils';
import { expect, it } from 'vitest';

export const testSnapShot = (wrapperGetter: () => VueWrapper) => {
  it('Matches the snapshot', () => {
    expect(wrapperGetter().html()).toMatchSnapshot();
  });
};
