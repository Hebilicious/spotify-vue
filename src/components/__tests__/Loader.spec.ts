import Loader from "@/components/Loader.vue";
import { testSnapShot } from "@/utils/test.utils";
import { shallowMount, VueWrapper } from "@vue/test-utils";
import { afterEach, beforeEach, describe } from "vitest";

describe('Loader', () => {
    let wrapper: VueWrapper;

    beforeEach(() => {
        wrapper = shallowMount(Loader)
    })

    afterEach(() => wrapper.unmount())

    testSnapShot(() => wrapper);
});