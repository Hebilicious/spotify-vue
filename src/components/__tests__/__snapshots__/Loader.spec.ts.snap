// Vitest Snapshot v1

exports[`Loader > Matches the snapshot 1`] = `
"<div class=\\"loader-container\\">
  <div class=\\"loader\\"></div>
</div>"
`;
